.github/workflows/check_runner.yml

```diff
@@ -0,0 +1,45 @@
+name: Check Runner
+
+on:
+  workflow_call:
+    inputs:
+      primary-runner:
+        required: true
+        type: string
+      fallback-runner:
+        required: true
+        type: string
+
+    outputs:
+      selected-runner:
+        description: "The runner to use (primary if available, otherwise fallback)"
+        value: ${{ jobs.detect-runner.outputs.selected-runner }}
+
+jobs:
+  detect-runner:
+    runs-on: ubuntu-latest
```

Comment:
Shouldn't this be specified where you run this workflow? Currently extras is waiting for the macos runner